TestingLowerBounds/StatInfoFun.lean

```diff
@@ -0,0 +1,435 @@
+      _ = μ (Ι (β * x) β) * ENNReal.ofReal |β - β * x| := by
+        rw [uIoc_comm, uIoc_of_ge hβxβ, abs_of_nonneg (sub_nonneg.mpr hβxβ), mul_comm]
+  · have hββx : β * x ≤ β := by exact mul_le_of_le_one_right hβ hx
+    calc
+      _ ≤ ∫⁻ γ, ENNReal.ofReal ((Ioc (β * x) β).indicator (fun _ ↦ β - β * x) γ) ∂μ :=
+        lintegral_mono fun _ ↦ ENNReal.ofReal_le_ofReal <|
+          statInfoFun_le_of_nonneg_of_right_le_one hβ hx
+      _ = ∫⁻ γ,  ((Ioc (β * x) β).indicator (fun _ ↦ ENNReal.ofReal (β - β * x)) γ) ∂μ := by
+        simp [Set.comp_indicator]
+      _ ≤ ENNReal.ofReal (β - β * x) * μ (Ioc (β * x) β) := lintegral_indicator_const_le _ _
+      _ = μ (Ι (β * x) β) * ENNReal.ofReal |β - β * x| := by
+        rw [uIoc_comm, uIoc_of_ge hββx, abs_of_nonneg (sub_nonneg.mpr hββx), mul_comm]
+  · have hβxβ : β ≤ β * x := by exact le_mul_of_one_le_right hβ hx
+    calc
+      _ ≤ ∫⁻ γ, ENNReal.ofReal ((Ioc β (β * x)).indicator (fun _ ↦ β * x - β) γ) ∂μ :=
+        lintegral_mono fun _ ↦ ENNReal.ofReal_le_ofReal <|
+          statInfoFun_le_of_nonneg_of_one_le_right hβ hx
+      _ = ∫⁻ γ,  ((Ioc β (β * x)).indicator (fun _ ↦ ENNReal.ofReal (β * x - β)) γ) ∂μ := by
+        simp [Set.comp_indicator]
+      _ ≤ ENNReal.ofReal (β * x - β) * μ (Ioc β (β * x)) := lintegral_indicator_const_le _ _
+      _ = μ (Ι (β * x) β) * ENNReal.ofReal |β - β * x| := by
+        rw [uIoc_of_ge hβxβ, mul_comm, abs_sub_comm, abs_of_nonneg (sub_nonneg.mpr hβxβ)]
+
+lemma integrable_statInfoFun {μ : Measure ℝ} [IsLocallyFiniteMeasure μ] (β x : ℝ) :
+    Integrable (fun γ ↦ statInfoFun β γ x) μ := by
+  refine ⟨measurable_statInfoFun2.stronglyMeasurable.aestronglyMeasurable, ?_⟩
+  refine ((lintegral_nnnorm_statInfoFun_le _ _).trans_lt ?_)
+  refine ENNReal.mul_lt_top ?_ ENNReal.ofReal_ne_top
+  exact (measure_mono uIoc_subset_uIcc).trans_lt isCompact_uIcc.measure_lt_top |>.ne
+
+end statInfoFun_γ
+
+section fDiv
+
+lemma nnReal_mul_fDiv {a : NNReal} :
+    a * fDiv (fun x ↦ statInfoFun β γ x) μ ν = fDiv (statInfoFun (a * β) (a * γ)) μ ν := by
+  change (a.1 : EReal) * _ = _
+  rw [← fDiv_mul a.2 ((convexOn_statInfoFun β γ).subset (fun _ _ ↦ trivial) (convex_Ici 0)) μ ν]
+  simp_rw [const_mul_statInfoFun a.2]
+  rfl
+
+end fDiv
+
+section CurvatureMeasure
+
+--should we define this to be some junk value if f is not convex? this way we could avoid having to state the convexity every time
+-- this may be put in some other place, maybe directly in the stieltjes file
+/-- The curvature measure induced by a convex function. It is defined as the only measure that has
+the right derivative of the function as a CDF. -/
+noncomputable
+def curvatureMeasure (f : ℝ → ℝ) (hf : ConvexOn ℝ univ f) : Measure ℝ :=
+  hf.rightDerivStieltjes.measure
+
+instance (f : ℝ → ℝ) (hf : ConvexOn ℝ univ f) : IsLocallyFiniteMeasure (curvatureMeasure f hf) := by
+  unfold curvatureMeasure
+  infer_instance
+
+lemma generalized_taylor (hf : ConvexOn ℝ univ f) (hf_cont : Continuous f) {a b : ℝ} :
```

Comment:
Maybe `convex_taylor` to be slightly more specific?
Also, can you add a docstring to that lemma? You could also change `lemma` to `theorem`, to indicate that this is an important result.